SU2_CFD/src/drivers/CDiscAdjSinglezoneDriver.cpp

```diff
@@ -439,62 +439,86 @@ void CDiscAdjSinglezoneDriver::DirectRun(RECORDING kind_recording){
-      if (config->GetWeakly_Coupled_Heat()){
-        cout << "log10[Heat(0)]: "   << log10(solver[HEAT_SOL]->GetRes_RMS(0)) << "." << endl;
-      }
-      if ( config->AddRadiation()) {
-        cout <<"log10[E(rad)]: " << log10(solver[RAD_SOL]->GetRes_RMS(0)) << endl;
-      }
-      break;
+    const unsigned short fieldWidth = 12;
+    PrintingToolbox::CTablePrinter RMSTable(&std::cout);
 
-    case DISC_ADJ_FEM:
+    /*--- The CTablePrinter requires two sweeps:
+     *--- 0. Add the colum names (addVals=0=false) plus CTablePrinter.PrintHeader()
+     *--- 1. Add the RMS-residual values (addVals=1=true) plus CTablePrinter.PrintFooter() ---*/
+    for (int addVals = 0; addVals < 2; addVals++) {
+      if(config->GetFluidProblem() || config->GetFEMSolver()) {
+
+        for (unsigned short iVar = 0; iVar < solver[FLOW_SOL]->GetnVar(); iVar++) {
+          if (!addVals)
+            RMSTable.AddColumn(string("rms_Flow[") + std::to_string(iVar) + string("]"), fieldWidth);
+          else
+            RMSTable << log10(solver[FLOW_SOL]->GetRes_RMS(iVar));
+        }
+
+        if ( config->GetKind_Turb_Model() != NONE && !config->GetFrozen_Visc_Disc()) {
+          for (unsigned short iVar = 0; iVar < solver[TURB_SOL]->GetnVar(); iVar++) {
+            if (!addVals)
+              RMSTable.AddColumn(string("rms_Turb[") + std::to_string(iVar) + string("]"), fieldWidth);
+            else
+              RMSTable << log10(solver[TURB_SOL]->GetRes_RMS(iVar));
+          }
+        }
+
+        if (config->GetWeakly_Coupled_Heat()){
+          if (!addVals) RMSTable.AddColumn("rms_Heat[0]", fieldWidth);
+          else RMSTable << log10(solver[HEAT_SOL]->GetRes_RMS(0));
+        }
+
+        if (config->AddRadiation()) {
+          if (!addVals) RMSTable.AddColumn("log10[E(0)]", fieldWidth);
+          else RMSTable << log10(solver[RAD_SOL]->GetRes_RMS(0));
+        }
 
-      if (config->GetGeometricConditions() == STRUCT_DEFORMATION::LARGE){
-        cout << "UTOL-A: "   << log10(solver[FEA_SOL]->GetRes_FEM(0))
-             << ", RTOL-A: " << log10(solver[FEA_SOL]->GetRes_FEM(1))
-             << ", ETOL-A: " << log10(solver[FEA_SOL]->GetRes_FEM(2)) << "." << endl;
       }
-      else{
-        if (geometry->GetnDim() == 2){
-          cout << "log10[RMS Ux]: "   << log10(solver[FEA_SOL]->GetRes_RMS(0))
-               << ", log10[RMS Uy]: " << log10(solver[FEA_SOL]->GetRes_RMS(1)) << "." << endl;
+      else if (config->GetStructuralProblem()) {
+
+        if (config->GetGeometricConditions() == STRUCT_DEFORMATION::LARGE){
+          if (!addVals) {
+            RMSTable.AddColumn("UTOL-A", fieldWidth);
+            RMSTable.AddColumn("RTOL-A", fieldWidth);
+            RMSTable.AddColumn("ETOL-A", fieldWidth);
+          }
+          else {
+            RMSTable << log10(solver[FEA_SOL]->GetRes_FEM(0))
+                     << log10(solver[FEA_SOL]->GetRes_FEM(1))
+                     << log10(solver[FEA_SOL]->GetRes_FEM(2));
+          }
```

Comment:
I did not adapt the structural problems using the loop over iVar. Should I do that as well? I am not sure whether the GetnVar routine returns the correct amount

Reply:
Hmmm you can leave this one like that I think.